Updated Sequence views and URLs

Deep-Chill opened the pull request:
- Updated URLs to allow for EnumeratedElements with a negative integer rank.
- Removed is_live_event and is_certificate properties from the EnumeratedElements model and moved the logic to the views. This makes the integration of SurveyEvent in Djaopsp easier.

Deep-Chill committed:
[Work-in-Progress] Updated Sequence views and URLs

- Updated URLs to allow for negative integers, in-case there would be negative integer ranked EnumeratedElements.
- Removed is_live_event and is_certificate properties from the EnumeratedElements model and moved the logic to the views. This makes the integration of SurveyEvent in Djaopsp easier.

smirolo commented on pages/views/sequences.py:
```diff
@@ -80,7 +86,14 @@ def get_object(self, queryset=None):
-        return get_object_or_404(EnumeratedElements, sequence=sequence, rank=rank)
+        element = get_object_or_404(EnumeratedElements, sequence=sequence, rank=rank)
+
+        element.is_live_event = LiveEvent.objects.filter(element=element.page_element).exists()
```
Somehow it feels this code and the similar code above should be in a `decorate_element` method defined in a common mixin.

smirolo reviewed (approved):
What is the reason for negative ranks?

Deep-Chill committed:
Moves decorating element logic to Mixin

- Added SequenceProgressMixin with get_queryset and decorate_queryset methods

Deep-Chill committed:
Adds update_element method to SequenceProgressMixin

- Adds a helper function to help subclass SequenceProgressMixin in Djaopsp without having to loop through the queryset twice.

smirolo reviewed (approved):
Let's try it out!
